test/query_test.rb

```diff
@@ -50,4 +50,13 @@ def test_find_one!
     assert_equal user, User.find_one!(name: "Bryan")
     assert_raises(Minidoc::DocumentNotFoundError) { User.find_one!(name: "Noah") }
   end
+
+  def test_find_one_or_initialize
+    user = User.create(name: "Bryan")
+    assert_equal user, User.find_one_or_initialize({name: "Bryan"})
```

Comment:
💄 rubocop doesn't enforce it in this project, but we usually prefer whitespace inside hash braces. e.g. `{ name: "Bryan" }`.